ElrondNetwork/elrond-go
# Sync missing snapshot nodes

## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- If a trie node is missing, the node will not be able to progress. In order to fix this, when the state snapshot process is triggered, also start a sync process that will sync missing trie nodes from the network. 
  
## Proposed Changes
- Add a trie syncer to accountsDB. The syncer will be triggered if there are any missing nodes during the snapshot process.

## Testing procedure
- Check that everything works as before
- Do not save some trie nodes to storage, and check that they will be synced during snapshot
